src/main/java/org/fcrepo/transform/transformations/LDPathTransform.java

```diff
@@ -78,60 +76,57 @@ public LDPathTransform(final InputStream query) {
         this.query = query;
     }
 
-    /**
-     * Pull a node-type specific transform out of JCR
-     * @param node the node
-     * @param key the key
-     * @return node-type specific transform
-     * @throws RepositoryException if repository exception occurred
-     */
-    public static LDPathTransform getNodeTypeTransform(final Node node,
-        final String key) throws RepositoryException {
-
-        final Node programNode = node.getSession().getNode(CONFIGURATION_FOLDER + key);
+    public static LDPathTransform getResourceTransform(final FedoraResource resource, final NodeService nodeService,
+            final String key) throws RepositoryException {
 
-        LOGGER.debug("Found program node: {}", programNode.getPath());
+        final FedoraResource transformResource =
+                nodeService.find(resource.getNode().getSession(), CONFIGURATION_FOLDER + key);
 
-        final NodeType primaryNodeType = node.getPrimaryNodeType();
+        LOGGER.debug("Found transform resource: {}", transformResource.getPath());
 
-        final Set<NodeType> supertypes = orderedBy(BY_NAME).add(primaryNodeType.getSupertypes()).build();
-        final Set<NodeType> mixinTypes = orderedBy(BY_NAME).add(node.getMixinNodeTypes()).build();
+        final List<URI> rdfTypes = resource.getTypes();
 
-        // start with mixins, primary type, and supertypes of primary type
-        final ImmutableList.Builder<NodeType> nodeTypesB = builder();
-        nodeTypesB.addAll(mixinTypes).add(primaryNodeType).addAll(supertypes);
+        LOGGER.debug("Discovered rdf types: {}", rdfTypes);
 
-        // add supertypes of mixins
-        mixinTypes.stream().map(mixin -> orderedBy(BY_NAME).add(mixin.getDeclaredSupertypes()).build())
-            .forEach(nodeTypesB::addAll);
+        final NamespaceRegistry nsRegistry = resource.getNode().getSession().getWorkspace().getNamespaceRegistry();
 
-        final List<NodeType> nodeTypes = nodeTypesB.build();
-
-        LOGGER.debug("Discovered node types: {}", nodeTypes);
-        for (final NodeType nodeType : nodeTypes) {
-            if (programNode.hasNode(nodeType.toString())) {
-                return new LDPathTransform(programNode.getNode(nodeType.toString())
-                                               .getNode(JCR_CONTENT)
-                                               .getProperty(JCR_DATA)
-                                               .getBinary().getStream());
+        // convert rdf:type with URI namespace to prefixed namespace
+        final Function<URI, String> namespaceUriToPrefix = x -> {
+            final String uriString = x.toString();
+            try {
+                for (final String namespace : Arrays.asList(nsRegistry.getURIs())) {
+                    // There is an empty namespace URI and that matches everything.
```

Comment:
Perhaps my test is bad, but for some reason the namespace registry has a blank prefix as the first entry. That was matching. I'll take another look at it.